### ISSUE TITLE
[AIR 51.0.0.3][macOS] `Worker::start()` cause crash

### ISSUE THREAD

**Issue by itlancer** [airsdk/Adobe-Runtime-Support#3102]
## Problem Description
`Worker::start()` cause application crash using AIR 51.0.0.3 with macOS.

Tested with latest AIR 51.0.0.3 with macOS Intel-based device with different applications.
There is no such issue with Windows.
Also there is no such issue using AIR 50.2.4.4.
Didn't test with AIR 51.0.0.2.

Related issue: https://github.com/airsdk/Adobe-Runtime-Support/issues/2523

## Steps to Reproduce
Launch application with code below on any macOS device.

Application example with sources attached.
[air51_mac_worker_start_crash.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/14487375/air51_mac_worker_start_crash.zip)

```
package {
	import flash.display.Sprite;
	import flash.system.Worker;
	import flash.system.WorkerDomain;
	import flash.events.Event;
	
	public class AIR51MacWorkerStartCrash extends Sprite {
		private var worker:Worker;
		
		public function AIR51MacWorkerStartCrash() {
			if (Worker.current.isPrimordial) {
				trace("main");
				addEventListener(Event.ADDED_TO_STAGE, init);
			} else {
				trace("worker");
			}
		}
		
		private function init(e:Event):void {
			removeEventListener(Event.ADDED_TO_STAGE, init);
			worker = WorkerDomain.current.createWorker(this.loaderInfo.bytes, true);
			worker.start();//This line cause crash
		}
	}
}
```

**Actual Result:**
Application crash:
```
Process:               mac_worker_crash [2974]
Path:                  /Volumes/VOLUME/*/mac_worker_crash.app/Contents/MacOS/mac_worker_crash
Identifier:            mac.worker.crash
Version:               1.0.0 (1.0.0)
Code Type:             X86-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2024-03-04 23:20:18.6108 +0300
OS Version:            macOS 13.1 (22C65)
Report Version:        12
Bridge OS Version:     7.1 (20P2059)
Anonymous UUID:        1349E91B-F18E-259F-0DC9-E53B57BEAB1F

Sleep/Wake UUID:       88FC3CA2-5681-4D42-8B83-257797E24D27

Time Awake Since Boot: 3600 seconds
Time Since Wake:       1367 seconds

System Integrity Protection: enabled

Notes:
dyld_process_snapshot_create_for_process failed with 5

Crashed Thread:        19

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000

Application Specific Backtrace 0:
0   CoreFoundation                      0x00007ff81a75940b __exceptionPreprocess + 242
1   libobjc.A.dylib                     0x00007ff81a2a5e25 objc_exception_throw + 48
2   CoreFoundation                      0x00007ff81a7815b6 _CFBundleGetValueForInfoKey + 0
3   AppKit                              0x00007ff81d851584 -[NSWindow(NSWindow_Theme) _postWindowNeedsToResetDragMarginsUnlessPostingDisabled] + 307
4   AppKit                              0x00007ff81d83e353 -[NSWindow _initContent:styleMask:backing:defer:contentView:] + 1247
5   AppKit                              0x00007ff81d9d9072 -[NSPanel _initContent:styleMask:backing:defer:contentView:] + 50
6   AppKit                              0x00007ff81d83de6d -[NSWindow initWithContentRect:styleMask:backing:defer:] + 42
7   AppKit                              0x00007ff81d9d902b -[NSPanel initWithContentRect:styleMask:backing:defer:] + 59
8   AppKit                              0x00007ff81d83c38b -[NSWindowTemplate nibInstantiate] + 354
9   AppKit                              0x00007ff81d80a822 -[NSIBObjectData instantiateObject:] + 222
10  AppKit                              0x00007ff81d809f94 -[NSIBObjectData nibInstantiateWithOwner:options:topLevelObjects:] + 476
11  AppKit                              0x00007ff81d7fea6f loadNib + 420
12  AppKit                              0x00007ff81d7fde57 +[NSBundle(NSNibLoading) _loadNibFile:nameTable:options:withZone:ownerBundle:] + 737
13  AppKit                              0x00007ff81d7fda81 -[NSBundle(NSNibLoading) loadNibNamed:owner:topLevelObjects:] + 201
14  AppKit                              0x00007ff81db50c60 -[NSAlert init] + 101
15  Adobe AIR                           0x0000000103bebc77 startAIR + 5304476
16  Adobe AIR                           0x0000000103e3ea77 startAIR + 7741084
17  Adobe AIR                           0x0000000103e00418 startAIR + 7485501
18  Adobe AIR                           0x0000000103e00f63 startAIR + 7488392
19  Adobe AIR                           0x0000000103a92338 startAIR + 3888989
20  Adobe AIR                           0x0000000103a92a14 startAIR + 3890745
21  Adobe AIR                           0x0000000103a420ce startAIR + 3560691
22  Adobe AIR                           0x0000000103a3e502 startAIR + 3545383
23  libsystem_pthread.dylib             0x00007ff81a605259 _pthread_start + 125
24  libsystem_pthread.dylib             0x00007ff81a600c7b thread_start + 15
```

**Expected Result:**
Worker will be started without crash.

## Known Workarounds
none
*Use AIR 50.2.4.4

**Comment by ajwfrost** [comment]
Ah...
```
Crashed Thread:        19
14  AppKit                              0x00007ff81db50c60 -[NSAlert init] + 101
```

We can't show an alert from a background thread. I think this is the worker instantiation, mistakenly trying to show an alert to say that it's a beta version of AIR....

thanks

**Comment by itlancer** [comment]
Fixed with latest AIR SDK 51.0.0.4.
Thanks!

**Comment by esdebon** [comment]
I have exactly the  same problem.

The log say:

 **Unsupported enumeration of UIWindowScene windows on non-main thread.**


![error](https://github.com/airsdk/Adobe-Runtime-Support/assets/1408296/aeba0894-97e9-4d45-8d95-c9770d12b24f)


> Unsupported enumeration of UIWindowScene windows on non-main thread.

> +[UIView setAnimationsEnabled:] being called from a background thread. Performing any operation from a background thread on UIView or a subclass is not supported and may result in unexpected and insidious behavior. trace=(
> 	0   UIKitCore                           0x00000001a22975f4 EEA1886F-3227-31C9-B7D8-202AF6530CFF + 18126324
> 	1   libdispatch.dylib                   0x00000001a66f4cc0 8A21415C-51AF-3AFC-8BA7-7EC2D9F7EEF1 + 404672
> 	2   libdispatch.dylib                   0x00000001a6697810 8A21415C-51AF-3AFC-8BA7-7EC2D9F7EEF1 + 22544
> 	3   UIKitCore                           0x00000001a118c648 EEA1886F-3227-31C9-B7D8-202AF6530CFF + 255560
> 	4   UIKitCore                           0x00000001a159e330 EEA1886F-3227-31C9-B7D8-202AF6530CFF + 4522800
> 

> *** Assertion failure in -[CTApplication _performAfterCATransactionCommitsWithLegacyRunloopObserverBasedTiming:block:], UIApplication.m:3225


> *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Call must be made on main thread'
> *** First throw call stack:
> (0x19f0c10a8 0x1972a6ca4 0x19e582640 0x1a1277704 0x1a1321720 0x1a11cf618 0x1a11ce5e0 0x1a12abd10 0x1a12abbc0 0x1a1484e48 0x1a1a5b2dc 0x1a1a5b28c 0x1a1a5b23c 0x1a1484db8 0x1a1484c58 0x1a1a5b1ec 0x1a159cb2c 0x1a159c764 0x1a159bd80 0x1a159e374 0x100e432f0 0x100ecc280 0x100ecc8a0 0x100ed3330 0x100ed378c 0x100ed3b0c 0x100ed9864 0x100eda264 0x101214894 0x101214e18 0x101353838 0x1012118b4 0x1f3e457a4 0x1f3e45134)

My code:

```
[Embed(source="./BackWorker.swf", mimeType="application/octet-stream")]
private static var BackgroundWorker:Class;
.
.
.
WorkerManager.worker = WorkerDomain.current.createWorker(( new BackgroundWorker() as ByteArray));
WorkerManager.worker.addEventListener(flash.events.Event.WORKER_STATE, statusWorker);
WorkerManager.bm = worker.createMessageChannel(Worker.current);
WorkerManager.mb = Worker.current.createMessageChannel(worker);
WorkerManager.worker.setSharedProperty("btm", WorkerManager.bm);
WorkerManager.worker.setSharedProperty("mtb", WorkerManager.mb);
WorkerManager.bm.addEventListener(flash.events.Event.CHANNEL_MESSAGE, onBackToMain);
WorkerManager.worker.start(); // Here Crash

```
To test I created an empty "BackWorker.swf" file and I got the same result, it always happens when calling the start method.

Where get the IR SDK 51.0.0.4??
